cpp/src/io/parquet/writer_impl.cu

```diff
@@ -919,7 +909,9 @@ void writer::impl::write(table_view const &table)
   if (fragments.size() != 0) {
     // Move column info to device
     col_desc.host_to_device(stream);
-    leaf_column_views = create_leaf_column_device_views(col_desc, *parent_column_table_device_view);
+    auto col_desc_span = cudf::detail::device_span<gpu::EncColumnDesc>{col_desc};
+    leaf_column_views  = cudf::io::create_leaf_column_device_views(
+      col_desc_span, *parent_column_table_device_view, stream);
```

Comment:
You should be able to pass the `hostdevice_vector` and it will be implicitly converted into `device_span`.
```suggestion
    leaf_column_views  = cudf::io::create_leaf_column_device_views(
      col_desc, *parent_column_table_device_view, stream);
```